chanzuckerberg/single-cell-data-portal
# feat: write to versioned snapshot paths when constructing the marker genes cube [wip]

## Reason for Change

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/5166

## Changes

- if `snapshot_schema_version` is provided when loading the cube, then we'll upload the snapshot artifacts to `s3://<bucket_name>/snapshots/<schema_version>/`
- introduces a default param of `read_versioned_snapshots` into `load_snapshot`. this is set to `True` in the `calculate_markers` pipeline, to ensure that we're writing to the new location. however, it's defaulted to `False` so that all of our API's that call this function continue reading from the old location at the root directory

## Testing steps

- run WMG pipeline in dev after merging
- verify that the metadata got written to the new path in S3

## Notes for Reviewer
